src/libexpr/flake/flake.cc

```diff
@@ -15,6 +15,61 @@ namespace flake {
 typedef std::pair<fetchers::Tree, FlakeRef> FetchedFlake;
 typedef std::vector<std::pair<FlakeRef, FetchedFlake>> FlakeCache;
 
+void setOverride(FlakeInputs & overrides, const InputPath & path, const FlakeRef & ref) {
+    // find/emplace for each existing/non-existing node
+    auto pathIter = path.begin();
+    auto currentNode = &(overrides.emplace(*pathIter, FlakeInput()).first->second);
+    ++pathIter;
+    while (pathIter != path.end()) {
+        currentNode = &(currentNode->overrides.emplace(*pathIter, FlakeInput()).first->second);
+        ++pathIter;
+    }
+    currentNode->ref = ref;
+}
+
+std::string printOverrides(const FlakeInputs & overrides, const InputPath & inputPathPrefix) {
```

Comment:
Please add a comment describing what the output of `printOverrides()` looks like.

Reply:
I changed that method a good bit, let me know if it still needs to be clarified